master/internal/grpcutil/api.go

```diff
@@ -29,12 +30,13 @@ const jsonPretty = "application/json+pretty"
 
 // NewGRPCServer creates a Determined gRPC service.
 func NewGRPCServer(db *db.PgDB, srv proto.DeterminedServer, enablePrometheus bool,
-	extConfig *model.ExternalSessions,
+	extConfig *model.ExternalSessions, logStore *logger.LogBuffer,
 ) *grpc.Server {
 	// In go-grpc, the INFO log level is used primarily for debugging
 	// purposes, so omit INFO messages from the master log.
 	logger := logrus.New()
 	logger.SetLevel(logrus.WarnLevel)
+	logger.AddHook(logStore)
```

Comment:
this will add other messages besides panics in

```
time="2023-09-21T16:32:29-04:00" level=error msg="finished unary call with code Internal" error="rpc error: code = Internal desc = D" grpc.code=Internal grpc.method=GetExperiments grpc.service=determined.api.v1.Determined grpc.start_time="2023-09-21T16:32:29-04:00" grpc.time_ms=0.597 span.kind=server system=grpc
```

but I think this is intended from the above comment, if we don't want these messages we shouldn't be logging them to stdout/stderr

stdout/stderr == masterLogs in my opinion